scripts/prepare_dolly.py

```diff
@@ -0,0 +1,133 @@
+"""Implementation derived from https://github.com/tloen/alpaca-lora"""
+import sys
+from pathlib import Path
+
+# support running without installing as a package
+wd = Path(__file__).parent.parent.resolve()
+sys.path.append(str(wd))
+
+import torch
+import requests
+import json
+from torch.utils.data import random_split
+from lit_llama.tokenizer import Tokenizer
+from tqdm import tqdm
+
+
+DATA_FILE = "https://raw.githubusercontent.com/databrickslabs/dolly/master/data/databricks-dolly-15k.jsonl"
+DATA_FILE_NAME = "dolly_data_cleaned.json"
+IGNORE_INDEX = -1
+
+
+def prepare(
+    destination_path: Path = Path("data/dolly"), 
+    tokenizer_path: Path = Path("checkpoints/lit-llama/tokenizer.model"),
+    test_split_size: int = 2000,
+    max_seq_length: int = 256,
+    seed: int = 42,
+    mask_inputs: bool = False,  # as in alpaca-lora
+) -> None:
+    """Prepare the Alpaca dataset for instruction tuning.
+    
+    The output is a training and validation dataset saved as `train.pt` and `val.pt`,
+    which stores the preprocessed and tokenized prompts and labels.
+    """
+    
+    destination_path.mkdir(parents=True, exist_ok=True)
+    file_path = destination_path / DATA_FILE_NAME
+    download(file_path)
+
+    # TODO: If we don't have the Meta weights, where do we get the tokenizer from?
+    tokenizer = Tokenizer(tokenizer_path)
+
+    with open(file_path, "r") as file:
+        data = file.readlines()
+        data = [json.loads(line) for line in data]
+    for item in data:
```

Comment:
the original dolly dataset contains - `instruction`, `context`, `category` and `response` keys.

I have mapped context-> input and response -> output based on the Alpaca dataset. `category` contains the info of the type of instruction and response. Not sure how valuable this information could be for fine-tuning, but we can append this in the prompt type of task.

After preparing the data using this script, users can directly use the `finetune_lora.py` script by providing the path to this dataset.

Reply:
Not sure about `category`, maybe we can look at the original Dolly codebase whether they include it during finetuning?

Reply:
@lantiga I quickly checked their prompt for training [here](https://github.com/databrickslabs/dolly/blob/master/training/consts.py) and couldn't find an usage of category.


On reading their [blog](https://www.databricks.com/blog/2023/04/12/dolly-first-open-commercially-viable-instruction-tuned-llm), I find that the categories were created for 7 tasks and rewarding the labelers:

```
We set up a contest, where the top 20 labelers would get a big award. We also outlined 7 very specific tasks:
```

Reply:
It is not used for training: https://github.com/search?q=repo%3Adatabrickslabs%2Fdolly%20category&type=code